### PR TITLE
Adding cli driver

### PR DESCRIPTION
**Pull request by blast-hardcheese**
Looking for feedback on if this is a useful development tool, or if it's already provided by something else I haven't seen yet.

Usage:

```
cli ./codegen-sbt/src/sbt-test/codegen/test-compile/src/main/graphql/schema.graphql /tmp/foop/src/main/scala/example/Schema.scala --splitFiles true
```

I also wasn't able to find where in zio-config something like `--help` would be able to be registered, presumably that would only be satisfied by something like `zio-cli` once that reaches maturity. For this reason, I could drop the second commit that exposes underlying errors, as it's unrelated to the primary goal of providing a `cli` driver at the sbt console to aide in iteration.

### CODE REVIEW

**Comment by ghostdogpr** [comment]
Not sure if I would use it but why not, it doesn't hurt 😄

File: cli/src/main/scala/caliban/cli/Launch.scala [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import _root_.caliban.tools.Codegen.GenType
+import _root_.caliban.tools._
+import zio._
+import zio.config.{ read, ConfigDescriptor, ConfigSource, ReadError }
+import zio.console._
+
+object Launch extends App {
+  def run(args: List[String]) = (for {
+    opts  <- ZIO.fromEither(Options.fromArgs(args))
+    files <- Codegen.generate(opts, GenType.Client)
```
Comment (paulpdaniels) [inline cli/src/main/scala/caliban/cli/Launch.scala:12]: So this wouldn't support server gen?

Comment (blast-hardcheese) [reply cli/src/main/scala/caliban/cli/Launch.scala]: Resolved, thanks for taking a look.

```
sbt> cli {client|schema} [opt [opt...]]
```

File: cli/src/main/scala/caliban/cli/Launch.scala [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+object Launch extends App {
+  val extractType: List[String] => IO[NoSuchElementException, (GenType, List[String])] = {
+    case "client" +: rest => UIO((GenType.Client, rest))
+    case "schema" +: rest => UIO((GenType.Schema, rest))
+    case args             => ZIO.fail(MissingGenType(args))
+  }
+
+  def mkDirs(path: Path): Task[Unit] =
+    Task(path.toFile.mkdirs).as(())
```
Comment (paulpdaniels) [inline cli/src/main/scala/caliban/cli/Launch.scala:22]: Hint: `.unit`

**Review by paulpdaniels — APPROVED** [review]

**Pull request closed without merging**